kolibri_explore_plugin/assets/src/views/SearchPage.vue

```diff
@@ -164,41 +166,58 @@
         return this.query.trim();
       },
       resultChannels() {
-        const { channels } = this.searchResult;
-        if (!channels || !channels.length) {
-          return null;
-        }
+        let allChannels = kinds.map((k) => {
```

Comment:
Using Javascript map/reduce/filter it should be possible to stay functional and do:
```
const allChannels = kinds.map(
    // map kinds to arrays of content nodes
  ).reduce(
    // reduce all arrays into a single array
  ).filter(
    // filter duplicates
  );
return allChannels;
```